### DISCUSSION TITLE
React-Hook-Form Example

### DISCUSSION THREAD

**Discussion by jacokok** [KevinVandy/material-react-table#387] [Q&A]
I am very interested in the todo: react-hook-form example.

I am able to register field with something like this.
```typescript
Edit: ({ cell, column, table }) => <TextField {...register("name",  { required: true})} defaultValue={cell.getValue()} placeholder="First name" />
```

But how would I call the handleSubmit function from react-hook-form?

Would appreciate any tips or ideas.

**Comment by lamhungypl** [comment]
Hi @jacokok  did you find a solution? .

I assume that we could combine `useFieldArray` with `columns`. Passing the `fieldId` to the `columnDef.meta.fieldId` and then render table row like

```
<Row  key={fieldID} .../>
```

**Reply by jacokok** [reply]
I ended up not using material react table for editing and adding of new records and only used it to display the grid.
